### DISCUSSION TITLE
Laravel 9.x using vendor language overrides changes default lang directory, resulting in no default translations anymore

### DISCUSSION THREAD

**Discussion by derMart** [laravel/framework#45148] [General]
Documentation for 9.x states that translations are stored in lang/[locale-id]/[group-id].php (https://laravel.com/docs/9.x/localization#introduction).  
Further it states, that to override specific translation keys one can place them into resources/lang/vendor/[vendor-id]/[locale-id]/[group-id].php (https://laravel.com/docs/9.x/localization#overriding-package-language-files)
So I would assume that lang/... keys are used if they are not overridden by resources/lang/vendor/...

Problem is, that as soon as I have some translations overwritten, laravel no longer uses lang/[locale-id]/... for default translations, but resources/lang/[locale-id]/... which breaks all default translation keys. I haven't found this behaviour documented in https://laravel.com/docs/9.x/localization but looking at the code one can confirm:

https://github.com/laravel/framework/blob/8bcd484c8648053acf3a12041e7c8294392e15f7/src/Illuminate/Foundation/Application.php#L326-L331

As long as resources/lang directory exists, it is used to search translations in and lang/ is not used anymore. This smells like a bug to me, as the documentation clearly states that default translation lives inside lang/ and not resources/lang. Am I missing something or should I create a bug issue?

best
Martin

**Comment by andrey-helldar** [comment]
I confirm that this is an unobvious mechanic. But it works this way to maintain backwards compatibility.

At the same time, Laravel 10 will also contain this mechanic:

https://github.com/laravel/framework/blob/d7cdf5ca69c2b8b1d44bf15cafecad64d13153a9/src/Illuminate/Foundation/Application.php#L326-L332

**Comment by mikethea1** [comment]
Just got burned by this issue and finding this post was a lifesaver. I think this oddity deserves an explicit mention in the docs.

**Comment by avinmaster** [comment]
+1!!!!
